video_core: Various small improvements and bug fixes

raphaelthegreat opened the pull request:
* Adds another case of missing shared memory barriers which fixes more LDS artifacts
* Improves texture cache overlap resolution to allow copying individual slices in a multi-layer mip-mapped image (credit to poly for testing and brainstorming)
* Fixes a crash when CE task finishes execution earlier than GFX task.
* Fixes crashes when emitting Quad/Rect tess shaders with default attributes

raphaelthegreat committed:
ir_passes: Add barrier at end of block too

raphaelthegreat committed:
vk_platform: Always assign names to resources

polybiusproxy committed:
texture_cache: Better overlap handling

raphaelthegreat committed:
liverpool: Avoid resuming ce_task when its finished

raphaelthegreat committed:
spirv_quad_rect: Skip default attributes

Fixes some crashes

raphaelthegreat committed:
memory: Improve buffer size clamping

raphaelthegreat committed:
liverpool: Relax binary header validity check

raphaelthegreat committed:
liverpool: Stub SetPredication with a warning

* Better than outright crash

raphaelthegreat committed:
emit_spirv: Implement round to zero mode

squidbus commented on src/shader_recompiler/backend/spirv/emit_spirv_quad_rect.cpp:
```diff
@@ -118,6 +118,10 @@ struct QuadRectListEmitter : public Sirit::Module {
         // Set attributes
         for (int i = 0; i < inputs.size(); i++) {
+            const auto& input = fs_info.inputs[i];
+            if (input.IsDefault()) {
```
I'm not clear on what the purpose of this, inputs are already not added to the vector when default: https://github.com/shadps4-emu/shadPS4/blob/bc0b42ee539113fdd4a8a41aca207f905dcd61d1/src/shader_recompiler/backend/spirv/emit_spirv_quad_rect.cpp#L280-L284

raphaelthegreat replied:
Yes but the loop here attempts to access them and crashes in OpAccessChain so we need to skip over them in similar fashion

squidbus replied:
Then I think the vector should just be fixed to not have bad slots and only contain the valid inputs, no need to do this check everywhere.

raphaelthegreat replied:
Done

raphaelthegreat committed:
liverpool: queue::pop takes the front element

raphaelthegreat committed:
image_info: Remove obsolete assert

The old code assumed the mip only had 1 layer thus a right overlap could not return mip 0. But with the new path we handle images that are both mip-mapped and multi-layer, thus this can happen

squidbus committed:
tile_manager: Fix size calculation

raphaelthegreat committed:
spirv_quad_rect: Skip default attributes
